src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -0,0 +1,448 @@
+  OUTCOME_TRY(auto&& exit_payload_function_address,
+              DlsymInTracee(pid, library_handle, kExitPayloadFunctionName));
+  process->exit_payload_function_address_ = absl::bit_cast<uint64_t>(exit_payload_function_address);
+
+  // Get memory, create the return trampoline and make it executable.
+  OUTCOME_TRY(auto&& return_trampoline_memory,
+              MemoryInTracee::Create(pid, 0, GetReturnTrampolineSize()));
+  process->return_trampoline_address_ = return_trampoline_memory->GetAddress();
+  auto result = CreateReturnTrampoline(pid, process->exit_payload_function_address_,
+                                       process->return_trampoline_address_);
+  OUTCOME_TRY(return_trampoline_memory->EnsureMemoryExecutable());
+
+  return process;
+}
+
+ErrorMessageOr<absl::flat_hash_set<uint64_t>> InstrumentedProcess::InstrumentFunctions(
+    const CaptureOptions& capture_options) {
+  OUTCOME_TRY(AttachAndStopProcess(pid_));
+  orbit_base::unique_resource detach_on_exit{pid_, [](int32_t pid) {
+                                               if (DetachAndContinueProcess(pid).has_error()) {
+                                                 ERROR("Detaching from %i", pid);
+                                               }
+                                             }};
+  // Init Capstone disassembler.
+  csh capstone_handle = 0;
+  cs_err error_code = cs_open(CS_ARCH_X86, CS_MODE_64, &capstone_handle);
+  if (error_code != CS_ERR_OK) {
+    return ErrorMessage("Failed to open Capstone disassembler.");
+  }
+  error_code = cs_option(capstone_handle, CS_OPT_DETAIL, CS_OPT_ON);
+  if (error_code != CS_ERR_OK) {
+    return ErrorMessage("Failed to configure Capstone disassembler.");
+  }
+  orbit_base::unique_resource close_on_exit{
+      &capstone_handle, [](csh* capstone_handle) { cs_close(capstone_handle); }};
+
+  OUTCOME_TRY(ExecuteInProcess(pid_, absl::bit_cast<void*>(start_new_capture_function_address_)));
+
+  OUTCOME_TRY(EnsureTrampolinesWritable());
+
+  absl::flat_hash_set<uint64_t> instrumented_function_ids;
+  for (const auto& function : capture_options.instrumented_functions()) {
+    const uint64_t function_id = function.function_id();
+    constexpr uint64_t kMaxFunctionPrologueBackupSize = 20;
+    const uint64_t backup_size = std::min(kMaxFunctionPrologueBackupSize, function.function_size());
+    if (backup_size == 0) {
+      // Can't instrument function of size zero
+      continue;
+    }
+    // Get all modules with the right path (usually one, but might be more) and get a function
+    // address to instrument for each of them.
+    OUTCOME_TRY(auto&& modules, ModulesFromModulePath(function.file_path()));
+    for (const auto& module : modules) {
+      // TODO: there is a function for that now, right? The page alignment is not handled here...
+      const uint64_t function_address =
+          module.address_start() + function.file_offset() - module.executable_segment_offset();
```

Comment:
`orbit_object_utils::SymbolVirtualAddressToAbsoluteAddress(function.file_offset(), module,address_start(), /*load_bias=*/0, module.executable_segment_offset());`

Reply:
actually it is better to introduce SymbolOffsetToAbsoluteAddress helper function which calls the one above with load_bias = 0